hy/core/macros.hy

```diff
@@ -20,6 +20,15 @@
          ~@(interleave (repeat name) rest))
      ~name))
 
+
+(defmacro assoc [coll k1 v1 &rest other-kvs]
+  (if (odd? (len other-kvs))
+    (macro-error (last other-kvs) "`assoc` takes an odd number of arguments"))
+  (setv c (if other-kvs (gensym "c") coll))
+  `(setv ~@(+ (if other-kvs [c coll] []) #*
+    (genexpr [`(get ~c ~k) v] [[k v] (partition (+ (, k1 v1) other-kvs))]))))
```

Comment:
It would also look better if `#*` were on the same line as the thing it's unpacking.